tests/Rule/RuleTest.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sabberworm\CSS\Tests\Value;
+
+use PHPUnit\Framework\TestCase;
+use Sabberworm\CSS\Parsing\ParserState;
+use Sabberworm\CSS\Settings;
+use Sabberworm\CSS\Rule\Rule;
+use Sabberworm\CSS\Value\RuleValueList;
+use Sabberworm\CSS\Value\ValueList;
+
+/**
+ * @covers \Sabberworm\CSS\Rule\Rule
+ */
+final class RuleTest extends TestCase
+{
+    /**
+     * @return array<string, array{0: string, 1: array<int, string>}>
+     */
+    public static function provideRulesAndExpectedParsedValueListTypes(): array
+    {
+        return [
+            'src (e.g. in @font-face)' => [
+                "
+                    src: url('../fonts/open-sans-italic-300.woff2') format('woff2'),
+                         url('../fonts/open-sans-italic-300.ttf') format('truetype');
+                ",
+                [RuleValueList::class, RuleValueList::class],
+            ],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @param array<int, string> $expectedTypeClassnames
+     *
+     * @dataProvider provideRulesAndExpectedParsedValueListTypes
+     */
+    public function parsesValuesIntoExpectedTypeList(string $rule, array $expectedTypeClassnames): void
+    {
+        $subject = Rule::parse(new ParserState($rule, Settings::create()));
+
+        $value = $subject->getValue();
+        self::assertInstanceOf(ValueList::class, $value);
+
+        $actualClassnames = \array_map(
+            function ($component): string {
```

Comment:
We can make the function static and add a type declaration for `$component`.

Reply:
Done.  On checking the type for `$component`, I found it could be a `Value` or a `string`, so have added an additional check.